### PR TITLE
Ncnews 3 - Created /api/topics handler

### PR DESCRIPTION
**Pull request by bathcoder**
1. Installed supertest, express, pg
2. Created app/model/controller/testfile
3. Handled /api/topics + test 
4. Made 404 error handler

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,8 +8,8 @@ If you are cloning this project locally, you will need to do the following tasks
 .env.development
 
 2. Ensure the files contain the follwing: 
-The .env.development file should contain the following script: PGDATABASE= <databse name here>
-The .env.test file should contain the following script: PGDATABSE= <test databse name here>
+The .env.development file should contain the following script: PGDATABASE=<databse name here>
+The .env.test file should contain the following script: PGDATABSE=<test databse name here>
```
Comment (daviidbartlett) [inline README.md:12]: this should have been in previous PR

File: controllers/news_controller.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const { fetchTopics } = require("../models/news_model.js");
+
+exports.getTopics = (req, res) => {
+  fetchTopics()
+    .then((topics) => {
+      console.log(topics);
```
Comment (daviidbartlett) [inline controllers/news_controller.js]: remove console.log

File: models/news_model.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+const db = require("../db/index.js");
+
+exports.fetchTopics = (()=>{
+  return db.query('SELECT * FROM topics;')
+  .then((topics) =>{
```
Comment (daviidbartlett) [inline models/news_model.js]: the thing that comes back from the request is not the topics. Be careful of your variable names.

File: README.md (hunk 1 above)
Comment (bathcoder) [reply README.md:12]: Oh yes Kev who reviewed me said to delete the spaces but not put in another pull request and just carry on. I should have probably deleted the spaces before pulling to main though?

File: package.json [hunk 4]
```diff
@@ -24,11 +24,15 @@
...
+    "pg-format": "^1.0.4",
+    "supertest": "^6.2.3"
   },
   "dependencies": {
     "dotenv": "^16.0.1",
-    "pg": "^8.7.3"
+    "express": "^4.18.1",
+    "node": "^17.7.2",
+    "pg": "^8.7.3",
+    "postgres": "^3.2.2"
```
Comment (daviidbartlett) [inline package.json]: why have you installed Postgres? what does this do??

Comment (bathcoder) [reply package.json]: Not sure, i was just having problems getting the db to be rquired in the test file, turns out it was the order that i was requiring files that was the issue, but i installed pg first thinking this was the problem?

File: models/news_model.js (hunk 3 above)
Comment (bathcoder) [reply models/news_model.js]: Is this something i should do for all the places where i have called topics? or just here in the model?

**Pull request merged**